crates/sui-node/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use std::{collections::BTreeMap, sync::Arc, time::Duration};
+use sui_config::NodeConfig;
+use sui_core::{
+    authority::{AuthorityState, AuthorityStore},
+    authority_active::{gossip::gossip_process, ActiveAuthority},
+    authority_client::NetworkAuthorityClient,
+};
+use sui_gateway::api::{RpcGatewayOpenRpc, RpcGatewayServer};
+use sui_gateway::json_rpc::JsonRpcServerBuilder;
+use sui_gateway::read_api::ReadApi;
+use sui_storage::IndexStore;
+use tracing::info;
+
+// TODO extract the important bits from AuthorityServer and FullNode so that we can have a single
+// unified node.
```

Comment:
A link to https://github.com/MystenLabs/sui/issues/2068 in the comment would not go amiss.